deno/lib/ZodError.ts

```diff
@@ -375,6 +379,10 @@ export const defaultErrorMap = (
         message = `Number must be less than ${
           issue.inclusive ? `or equal to ` : ``
         }${issue.maximum}`;
+      else if (issue.type === "date")
+        message = `Date must be smaller than or equal to ${new Date(
```

Comment:
Should use issue.inclusive to decide whether the "or equal to" should be added, as is done for other types.